python/pyarrow/tests/test_compute.py

```diff
@@ -1842,6 +1842,26 @@ def test_cast():
     assert pc.cast(arr, expected.type) == expected
 
 
+def test_identity_cast_dates():
+    dt = datetime.date(1990, 3, 1)
+    data = [[dt], [dt]]
+
+    schema = pa.schema([
+        ('date32', pa.date32()),
+        ('date64', pa.date64()),
+    ])
+
+    batch = pa.RecordBatch.from_arrays(data, schema=schema)
+
+    table = pa.RecordBatchReader.from_batches(
+        schema,
+        [batch]
+    ).cast(schema).read_all()
```

Comment:
Why do we need to create a table here?
Can we use `pc.cast()` or `Array.cast` like other tests do?

Reply:
Yes, certainly. I've been using this code since it was exactly the code that caused issues on my end. I've updated the test 👍